Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/EditorDisabledCompositionComponent.cpp

```diff
@@ -95,16 +127,18 @@ namespace AzToolsFramework
             // This is a special case for certain EditorComponents only!
             EditorDisabledCompositionRequestBus::Handler::BusConnect(GetEntityId());
 
-            // Set the entity* for each disabled component
-            for (auto disabledComponent : m_disabledComponents)
+            // Set the entity for each disabled component.
+            for (auto const& [componentAlias, disabledComponent] : m_disabledComponents)
             {
                 // It's possible to get null components in the list if errors occur during serialization.
                 // Guard against that case so that the code won't crash.
                 if (disabledComponent)
                 {
                     auto editorComponentBaseComponent = azrtti_cast<Components::EditorComponentBase*>(disabledComponent);
                     AZ_Assert(editorComponentBaseComponent, "Editor component does not derive from EditorComponentBase");
+
                     editorComponentBaseComponent->SetEntity(GetEntity());
+                    editorComponentBaseComponent->SetSerializedIdentifier(componentAlias);
```

Comment:
While looking at the custom serializer for composition components, I think it might be good to set up the serialized identifier over there instead and remove this line here.